### ISSUE TITLE
Function to get a filedescriptor for select()

### ISSUE THREAD

**Issue by honx** [eclipse-paho/paho.mqtt.c#667]
moin,

I'd need a function that returns a file descriptor for an mqtt connection that can be used for select() to integrate mqtt into an existig multiplexed communication system that handles a number of protocol.  (similar to the SSL_get_fd() function of openssl).

While i can add this internally it would of course be nicer to have it in the standard library. Would this functionality be something you'd integrate as a pull request?

**Comment by icraggs** [comment]
Yes I'd be open to this as a PR.

**Comment by clauderobi** [comment]
I guess the PR was never created but just in case, was it?

**Comment by honx** [comment]
No, sorry, because the project went a different way. Well not yet, actually, it may be that this gets implemented at a later time. If we do create something I'll be sure to submit a patch.
